src/gen.ts

```diff
@@ -391,12 +464,17 @@ export function processCategory(cursor: CXCursor, metadata: Metadata) {
 }
 
 export function processProtocol(cursor: CXCursor, metadata: Metadata) {
+  const avail = cursor.getPlatformAvailability();
+  if (avail.alwaysUnavailable || avail.alwaysDeprecated) return;
+
   const protocolDecl: ProtocolDecl = {
     name: cursor.getSpelling(),
+    file: cursor.getLocation().getFileLocation().file.getName(),
     protocols: [],
     properties: [],
     instanceMethods: [],
     classMethods: [],
+    availability: avail.availability,
   };
```

Comment:
nit: by destructuring `avail`, we can avoid the redundant-looking `avail.availability`.
```suggestion
  const { alwaysUnavailable, alwaysDeprecated, availability } =
    cursor.getPlatformAvailability();
  if (alwaysUnavailable || alwaysDeprecated) return;

  const protocolDecl: ProtocolDecl = {
    name: cursor.getSpelling(),
    file: cursor.getLocation().getFileLocation().file.getName(),
    protocols: [],
    properties: [],
    instanceMethods: [],
    classMethods: [],
    availability,
  };
```
(Same goes for the other three places where we call `cursor.getPlatformAvailability()`)